ldms/src/core/ldms_xprt.c

```diff
@@ -2081,17 +2082,18 @@ static void ldms_zap_handle_conn_req(zap_ep_t zep)
 
 	__ldms_xprt_conn_msg_init(x, &msg);
 
+	/* Take a 'connect' reference. Dropped in ldms_xprt_close() */
+	ldms_xprt_get(_x);
+
 	zerr = zap_accept(zep, ldms_zap_auto_cb, (void*)&msg, sizeof(msg));
 	if (zerr) {
 		x->log("ERROR: %d accepting connection from %s.\n", zerr, rmt_name);
```

Comment:
If `zap_accept()` returns a synchronous error, `zap_reject()` should not be called at line 2101.